index.js

```diff
@@ -108,7 +138,114 @@ function Jasmine2HTMLReporter(options) {
         return __specs[spec.id];
     }
 
-    self.jasmineStarted = function(summary) {
+    function getReportFilename(specName) {
+        var name = '';
+        if (self.fileNamePrefix)
+            name += self.fileNamePrefix + self.fileNameSeparator;
+
+        name += self.fileName;
+
+        if (specName !== undefined)
+            name += self.fileNameSeparator + specName;
+
+        if (self.fileNameSuffix)
+            name += self.fileNameSeparator + self.fileNameSuffix;
+
+        if (self.fileNameDateSuffix)
+            name += self.fileNameSeparator + getReportDate();
+
+        return name;
+    }
+
+    /**
+     * Remove passed specs but keep specs that failed on first run
+     * but passed on second.
+     * @param {object} suite - a suite object
+     */
+    function filterOutPassedSpecs(suite) {
+        var suiteName = getFullyQualifiedSuiteName(suite);
+
+        suite._specs = suite._specs.filter((spec) => {
+            return !isPassed(spec) || (flakedSuiteNames[suiteName] || []).indexOf(spec.id) > -1;
+        });
+
+        return suite;
+    }
+
+    /**
+     * Persist failed suite names to node local storage
+     */
+    function saveFlakedSuiteNames() {
+        if (Object.keys(flakedSuiteNames).length === 0)
+            return
+
+        var storedNames = loadFlakedSuiteNames();
+        namesToBeStored = Object.assign(storedNames, flakedSuiteNames);
+
+        storage.setItem(storageUID, JSON.stringify(namesToBeStored));
+    }
+
+    /**
+     * Retrieve flaked suite names from node local storage to determine if current
+     * run is a rerun.
+     */
+    function loadFlakedSuiteNames() {
+        var storedSuiteNames = storage.getItem(storageUID) || {};
+
+        if (typeof storedSuiteNames === 'string') {
+            return JSON.parse(storedSuiteNames);
```

Comment:
I would wrap this in a try/catch since JSON.parse can throw exceptions on malformed JSON.